sktime/forecasting/compose/_reduce.py

```diff
@@ -606,11 +705,33 @@ def _predict_last_window(
                 X_pred = X_pred.reshape(1, -1)
 
             # Allocate array for predictions.
-            y_pred = np.zeros(len(fh))
+            if est_type == "regressor":
+                y_pred = np.zeros(len(fh))
+            else:  # est_type == "regressor_proba"
+                y_preds = []
 
             # Iterate over estimators/forecast horizon
             for i, estimator in enumerate(self.estimators_):
-                y_pred[i] = estimator.predict(X_pred)
+                y_pred_est = getattr(estimator, method)(X_pred, **kwargs)
+                if est_type == "regressor":
+                    y_pred[i] = y_pred_est
+                else:  # est_type == "regressor_proba"
+                    y_pred_v = _coerce_to_numpy(y_pred_est)
+                    y_pred_i = _create_fcst_df([fh[i]], y_pred_est, fill=y_pred_v)
+                    y_preds.append(y_pred_i)
+
+            if est_type != "regressor":
+                y_pred = pool_preds(y_preds)
```

Comment:
Is there a reason, why it is not possible to use pre-allocated arrays here? Or why you haven't used them?

Reply:
the probabilistic predictions are 2D arrays, not 1D - so I would have to write extra code. This is hard due to `_create_fcst_df` which imo we should probably rewrite at some point.

I would prefer not to do array magic right now, and simplify things in a separate step, where we test that functionality is equal to this.

Unless you can see how to do things elegantly and easily.

Reply:
I think that extending it should not be that difficult, since we just need an extra dimension and then fill it as we do it currently. But doing it stepwise is probably better. Do we have an umbrella issue, where we could collect such issues?

Reply:
here is one, though perhaps we should open one for the existing reducer?
https://github.com/sktime/sktime/issues/3224

Regarding "should not be that difficult", with the existing reducer I thinkn it is. For insntance, `_create_fcst_df` - which alreay is in the current code - produces a data frame, then proceeds to fill it into a pre-allocated array. This looks strange and I hesitate to touch this outside of a complete rework of that location.

Reply:
My suggestion would be: merge this, and then we continue with debugging base on https://github.com/sktime/sktime/pull/5722 - the `"global"` parameter failing for "normal" forecasts.

Once that is fixed, we give it another pass and try to simplify the array stuff.

Reply:
That sounds as a plan.

Reply:
I would prefer a new issue that is linking together all the reducers related issues and also further open questions.

Reply:
Opened here: https://github.com/sktime/sktime/issues/5736

if something comes to your mind, please add.